doc/src/manual/code-loading.md

```diff
@@ -0,0 +1,321 @@
+
+**The depedency graph** of a project environment is determined by the contents of the manifest file, if present, or if there is no manifest file, `graph` is empty. A manifest file contains a stanza for each direct or indirect dependency of a project, including for each one, its UUID and exact version information and optionally an explicit path to to its source code. Consider the following example manifest file for `App`:
+
+```toml
+[[Priv]] # the private one
+deps = ["Pub", "Zebra"]
+uuid = "ba13f791-ae1d-465a-978b-69c3ad90f72b"
+path = "deps/Priv"
+
+[[Priv]] # the public one
+uuid = "2d15fe94-a1f7-436c-a4d8-07a9a496e01c"
+git-tree-sha1 = "1bf63d3be994fe83456a03b874b409cfd59a6373"
+version = "0.1.5"
+
+[[Pub]]
+uuid = "ba13f791-ae1d-465a-978b-69c3ad90f72b"
+git-tree-sha1 = "9ebd50e2b0dd1e110e842df3b433cb5869b0dd38"
+version = "2.1.4"
+
+  [Pub.deps]
+  Priv = "2d15fe94-a1f7-436c-a4d8-07a9a496e01c"
+  Zebra = "f7a24cb4-21fc-4002-ac70-f0e3a0dd3f62"
+
+[[Zebra]]
+uuid = "f7a24cb4-21fc-4002-ac70-f0e3a0dd3f62"
+git-tree-sha1 = "e808e36a5d7173974b90a15a353b564f3494092f"
+version = "3.4.2"
+```
+
+This manifest file describes a possible complete dependency graph for the `App` project:
+
+- There are two different `Priv` packages that the application needs—a private one which is a direct dependency and a public one which is an indirect dependency through `Pub`:
+  * The private `Priv` depends on the `Pub` and `Zebra` packages.
+  * The public `Priv` has no depdendencies.
+- The application also depends on the `Pub` package, which in turn depends on the public `Priv ` and the same `Zebra` package which the private `Priv` package depends on.
+
+A materialized representation of this dependency `graph` looks like this:
+
+```julia
+graph = Dict{UUID,Dict{Symbol,UUID}}(
+    # Priv – the private one:
+    UUID("ba13f791-ae1d-465a-978b-69c3ad90f72b") => Dict{Symbol,UUID}(
+        :Zebra => UUID("f7a24cb4-21fc-4002-ac70-f0e3a0dd3f62"),
+    ),
+    # Priv – the public one:
+    UUID("2d15fe94-a1f7-436c-a4d8-07a9a496e01c") => Dict{Symbol,UUID}(),
+    # Pub:
+    UUID("ba13f791-ae1d-465a-978b-69c3ad90f72b") => Dict{Symbol,UUID}(
+        :Priv  => UUID("2d15fe94-a1f7-436c-a4d8-07a9a496e01c"),
+        :Zebra => UUID("f7a24cb4-21fc-4002-ac70-f0e3a0dd3f62"),
+    ),
+    # Zebra:
+    UUID("f7a24cb4-21fc-4002-ac70-f0e3a0dd3f62") => Dict{Symbol,UUID}(),
+)
+```
+
+Again, for efficicency reasons, Julia doesn't actually materialize this graph during package loading, instead it computes portions of the graph as needed. Given this dependency `graph`, when Julia sees `import Priv` in the `Pub` package—which has UUID `ba13f791-ae1d-465a-978b-69c3ad90f72b`—it looks up:
```

Comment:
> Again, for efficicency reasons...

Same here as above.

Reply:
I’m just worried that people are going to go look for these APIs which don’t actually exist anywhere. Maybe they should, but currently they don’t.